src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf16Utility.Validation.cs

```diff
@@ -433,5 +479,27 @@ static Utf16Utility()
             scalarCountAdjustment = tempScalarCountAdjustment;
             return pInputBuffer;
         }
+
+        private static readonly Vector128<byte> s_mostSignficantBitMask = Vector128.Create((byte)0x80);
+        private static readonly Vector128<byte> s_bitMask128 = BitConverter.IsLittleEndian ?
+                                                Vector128.Create(0x80402010_08040201).AsByte() :
+                                                Vector128.Create(0x01020408_10204080).AsByte();
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static uint GetNonAsciiBytes(Vector128<byte> value)
+        {
+            Debug.Assert(AdvSimd.Arm64.IsSupported);
+
+            // extractedBits[i] = (value[i] & 0x80) == 0x80 & (1 << i);
+            Vector128<byte> mostSignficantBitMask = s_mostSignficantBitMask;
+            Vector128<byte> mostSignificantBitIsSet = AdvSimd.CompareEqual(AdvSimd.And(value, mostSignficantBitMask), mostSignficantBitMask);
+            Vector128<byte> extractedBits = AdvSimd.And(mostSignificantBitIsSet, s_bitMask128);
```

Comment:
Same suggestion as in https://github.com/dotnet/runtime/pull/39103#discussion_r454587164
This could be done with
```c#
Vector128<byte> mostSignificantBitIsSet = AdvSimd.ShiftRightArithmetic(value.AsSByte(), 7).AsByte();
```